sql/core/src/main/scala/org/apache/spark/sql/columnar/ColumnType.scala

```diff
@@ -34,7 +34,8 @@ import org.apache.spark.unsafe.types.UTF8String
  *
  * @tparam JvmType Underlying Java type to represent the elements.
  */
-private[sql] sealed abstract class ColumnType[JvmType] {
+private[sql]
+sealed abstract class ColumnType[@specialized(Boolean, Byte, Short, Int, Long) JvmType] {
```

Comment:
Checked the with `javap`, and it turned out that `ColumnType.copyField` always calls the boxed version of `setField` and `getField`. I think we should revert this part of changes since it's somewhat tricky to get optimal byte code with `@specialized`.